src/protocol/thread/thread.ts

```diff
@@ -362,12 +366,22 @@ class _ThreadFront {
     range?: { start: SourceLocation; end: SourceLocation }
   ) {
     assert(this.sessionId);
+
+    if (this.breakpointPositions.has(sourceId)) {
+      return this.breakpointPositions.get(sourceId);
+    }
+
     const begin = range ? range.start : undefined;
     const end = range ? range.end : undefined;
     const { lineLocations } = await client.Debugger.getPossibleBreakpoints(
       { sourceId, begin, end },
       this.sessionId
     );
+
+    if (!range) {
+      this.breakpointPositions.set(sourceId, lineLocations);
```

Comment:
Can we store a promise in the map so that if there are multiple requests for the breakpoint positions before the first request finishes, we still only ask the server for the positions once?